osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -112,21 +112,19 @@ private void load(OsuConfigManager config)
         private void updateItems()
         {
             skinItems = skins.GetAllUsableSkins();
-
-            // insert after lazer built-in skins
-            int firstNonDefault = skinItems.FindIndex(s => s.ID > 0);
-            if (firstNonDefault < 0)
-                firstNonDefault = skinItems.Count;
-
-            skinItems.Insert(firstNonDefault, random_skin_info);
-
+            skinItems = sortList(skinItems);
             skinDropdown.Items = skinItems;
         }
 
         private void itemUpdated(ValueChangedEvent<WeakReference<SkinInfo>> weakItem)
         {
             if (weakItem.NewValue.TryGetTarget(out var item))
-                Schedule(() => skinDropdown.Items = skinDropdown.Items.Where(i => !i.Equals(item)).Append(item).ToArray());
```

Comment:
The `.Where(i => !i.Equals(item))` check has been incorrectly removed. `itemUpdated` is also fired on skin updates, not necessarily always additions.